addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1319,7 +1336,7 @@ const InputUserValueWidget = UnitUserValueWidget.extend({
      */
     _onInputKeydown: function (ev) {
         const params = this._methodsParams;
-        if (!params.unit && !params.step) {
+        if (!this._hasUnit() && !params.step) {
```

Comment:
In fact, now that I see this one, I wonder if the places where you do "if has unit" should not also consider when a `step` only is defined? From the steps to reproduce you wrote, don't you have the same bug with widgets where you define a step without an unit (we should have a few already)? That indicates it is a number (the up and down key work). Maybe instead of `hasUnit` it should be `isNumeric` (if unit, fake unit or step)?

And it probably should be used at this location too: `const useNumberAlignment = !!step || !!unit || !!this.el.dataset.fakeUnit || !!this.el.dataset.hideUnit;` ?

To investigate :pray: